artifactory/system.go

```diff
@@ -33,6 +40,1112 @@ func (v Versions) String() string {
+	EmailAttribute           *string            `yaml:"emailAttribute,omitempty" xml:"emailAttribute,omitempty"`
+	LdapPoisoningProtection  *bool              `yaml:"ldapPoisoningProtection,omitempty" xml:"ldapPoisoningProtection,omitempty"`
+	LdapUrl                  *string            `yaml:"ldapUrl,omitempty" xml:"ldapUrl,omitempty"`
+	Search                   *LdapSettingSearch `yaml:"search,omitempty" xml:"search,omitempty"`
+	UserDnPattern            *string            `yaml:"userDnPattern,omitempty" xml:"userDnPattern,omitempty"`
+	AllowUserToAccessProfile *bool              `yaml:"allowUserToAccessProfile,omitempty" xml:"allowUserToAccessProfile,omitempty"`
+	AutoCreateUser           *bool              `yaml:"autoCreateUser,omitempty" xml:"autoCreateUser,omitempty"`
+	Enabled                  *bool              `yaml:"enabled,omitempty" xml:"enabled,omitempty"`
+}
+
+// LdapSettingSearch represents the Search setting in an LDAPSetting.
+//
+type LdapSettingSearch struct {
+	ManagerDn       *string `yaml:"managerDn,omitempty" xml:"managerDn,omitempty"`
+	ManagerPassword *string `yaml:"managerPassword,omitempty" xml:"managerPassword,omitempty"`
+	SearchBase      *string `yaml:"searchBase,omitempty" xml:"searchBase,omitempty"`
+	SearchFilter    *string `yaml:"searchFilter,omitempty" xml:"searchFilter,omitempty"`
+	SearchSubTree   *bool   `yaml:"searchSubTree,omitempty" xml:"searchSubTree,omitempty"`
+}
+
+// LdapSettings is an alias for a slice of LdapSetting that can be
+// properly marshaled to/from YAML.
+type LdapSettings []*LdapSetting
+
+// MarshalYAML implements the yaml.Marshaller interface for LdapSettings.
+func (l LdapSettings) MarshalYAML() (interface{}, error) {
+	ldapSettingsMap := make(map[string]*LdapSetting)
+	for _, ldapSetting := range l {
+		if *ldapSetting == (LdapSetting{Key: ldapSetting.Key}) {
+			ldapSettingsMap[*ldapSetting.Key] = nil
+		} else {
+			ldapSettingsMap[*ldapSetting.Key] = ldapSetting
+		}
+	}
+
+	return ldapSettingsMap, nil
+}
+
+// UnmarshalYAML implements yaml.Unmarshaler for LdapSettings.
+func (l *LdapSettings) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	ldapSettingsMap := make(map[string]*LdapSetting)
+	if err := unmarshal(ldapSettingsMap); err != nil {
+		return err
+	}
+
+	var ldapSettingsSlice LdapSettings
+	for ldapSettingKey, ldapSetting := range ldapSettingsMap {
+		if ldapSetting == nil {
+			continue
+		}
+		ldapSetting.Key = &ldapSettingKey
+		ldapSettingsSlice = append(ldapSettingsSlice, ldapSetting)
+	}
+
+	*l = ldapSettingsSlice
+	return nil
+}
+
+// LdapGroupSetting represents the LDAP Group settings in Artifactory Security Configuration.
+//
+// Docs: https://www.jfrog.com/confluence/display/RTF/YAML+Configuration+File#YAMLConfigurationFile-Security(Generalsecurity,PasswordPolicy,LDAP,SAML,OAuth,HTTPSSO,Crowd)
+type LdapGroupSetting struct {
+	Name                 *string `yaml:"-" xml:"name,omitempty"`
```

Comment:
I noticed we're not allowing this field to be serialized in YAML due to the `-` tag.

This peaked my curiosity so was this intentional?

Reply:
Similar to [comment](https://github.com/target/go-arty/pull/57#discussion_r436119180).